zk_toolbox/crates/zk_supervisor/src/commands/sql_fmt.rs

```diff
@@ -0,0 +1,197 @@
+
+use anyhow::{bail, Context, Result};
+use common::spinner::Spinner;
+use sqruff_lib::{api::simple::get_simple_config, core::linter::core::Linter};
+use xshell::Shell;
+
+use super::lint_utils::{get_unignored_files, IgnoredData, Target};
+use crate::messages::{
+    msg_file_is_not_formatted, MSG_FAILED_TO_DETERMINE_BASE_INDENT,
+    MSG_FAILED_TO_FIND_END_OF_REGULAR_STRING_QUERY,
+    MSG_FAILED_TO_FIND_START_OF_REGULAR_STRING_QUERY, MSG_RUNNING_SQL_FMT_SPINNER,
+};
+
+fn format_query(query: &str) -> anyhow::Result<String> {
+    let exclude_rules = vec!["LT12".to_string()]; // avoid adding newline before `$` character
+    let cfg = get_simple_config(Some("postgres".into()), None, Some(exclude_rules), None).unwrap();
+    let mut linter = Linter::new(cfg, None, None);
+    let mut result = linter.lint_string_wrapped(query, None, true);
+    let formatted_query = take(&mut result.paths[0].files[0]).fix_string();
+    // Remove first empty line
+    let formatted_query = formatted_query
+        .lines()
+        .skip(1)
+        .collect::<Vec<&str>>()
+        .join("\n");
+
+    Ok(formatted_query)
+}
+
+fn extract_query_from_rust_string(query: &str, is_raw: bool) -> String {
+    let mut query = query.trim().to_string();
+    if query.ends_with(',') {
+        query.pop();
+    }
+    // Removing quotes
+    if !is_raw {
+        query = query[1..query.len() - 1].to_string();
+    } else {
+        query = query[3..query.len() - 2].to_string();
+    }
+
+    // Remove all escape characters
+    if !is_raw {
+        query = query.replace(r#"\""#, "\"");
+    }
+
+    query
+}
+
+fn embed_text_inside_rust_string(query: &str) -> String {
+    format!("r#\"\n{}\n\"#", query)
+}
+
+fn add_indent(query: &str, indent: usize) -> String {
+    query
+        .lines()
+        .map(|line| format!("{:indent$}{}", "", line))
+        .collect::<Vec<String>>()
+        .join("\n")
+}
+
+fn format_rust_string_query(query: &str, is_raw: bool) -> anyhow::Result<String> {
+    let base_indent = query.find(|c: char| !c.is_whitespace()).unwrap_or(0);
+    let raw_query = extract_query_from_rust_string(query, is_raw);
+    let formatted_query = format_query(&raw_query)?;
+    let reconstructed_rust_string = embed_text_inside_rust_string(&formatted_query);
+    Ok(add_indent(&reconstructed_rust_string, base_indent))
+}
+
+fn format_one_line_query(line: &str) -> anyhow::Result<String> {
+    let is_raw_string = line.contains("sqlx::query!(r");
+
+    let query_start = if is_raw_string {
+        line.find(r#"r#""#)
+            .context(MSG_FAILED_TO_FIND_START_OF_REGULAR_STRING_QUERY)?
+    } else {
+        line.find('"')
+            .context(MSG_FAILED_TO_FIND_START_OF_REGULAR_STRING_QUERY)?
+    };
+
+    let base_indent = line
+        .find(|c: char| !c.is_whitespace())
+        .context(MSG_FAILED_TO_DETERMINE_BASE_INDENT)?
+        + 4;
+
+    let prefix = &line[..query_start];
+
+    let query_end = if is_raw_string {
+        line.find(r#""#)
```

Comment:
Does it work like that? IIUC, `r#""#` is an empty string; you'd probably want `"\"#"`.